rs/pocket_ic_server/src/pocket_ic.rs

```diff
@@ -564,54 +561,64 @@ impl PocketIcSubnets {
         instruction_config: SubnetInstructionConfig,
         registry_data_provider: Arc<ProtoRegistryDataProvider>,
         create_at_registry_version: RegistryVersion,
-        nonmainnet_features: &NonmainnetFeatures,
+        icp_config: &IcpConfig,
         log_level: Option<Level>,
         bitcoin_adapter_uds_path: Option<PathBuf>,
     ) -> StateMachineBuilder {
         let subnet_type = conv_type(subnet_kind);
         let subnet_size = subnet_size(subnet_kind);
         let mut subnet_config = SubnetConfig::new(subnet_type);
-        // using `let NonmainnetFeatures { }` with explicit field names
-        // to force an update after adding a new field to `NonmainnetFeatures`
-        let NonmainnetFeatures {
-            enable_beta_features,
-            disable_canister_backtrace,
-            disable_function_name_length_limits,
-            disable_canister_execution_rate_limiting,
-        } = nonmainnet_features;
-        // using `EmptyConfig { }` explicitly
-        // to force an update after adding a new field to `EmptyConfig`
-        let mut hypervisor_config = if let Some(EmptyConfig {}) = enable_beta_features {
-            crate::beta_features::hypervisor_config()
-        } else {
-            execution_environment::Config::default()
+        // using `let IcpConfig { }` with explicit field names
+        // to force an update after adding a new field to `IcpConfig`
+        let IcpConfig {
+            beta_features,
+            canister_backtrace,
+            function_name_length_limits,
+            canister_execution_rate_limiting,
+        } = icp_config;
+        let mut hypervisor_config = match beta_features.clone().unwrap_or(IcpConfigFlag::Disabled) {
+            IcpConfigFlag::Disabled => execution_environment::Config::default(),
+            IcpConfigFlag::Enabled => crate::beta_features::hypervisor_config(),
+        };
+        match canister_backtrace {
```

Comment:
Don't we just want to keep the same logic as above?

```suggestion
        match canister_backtrace.cloen().unwrap_or(IcpConfigFlag::Enabled) {
```

Reply:
Makes sense! I've resolved my comments